src/test/java/hudson/plugins/mercurial/SCMTestBase.java

```diff
@@ -746,6 +746,39 @@ public Proc onLaunch(Launcher.ProcStarter p) throws IOException {
         assertEquals(Result.SUCCESS, b.getResult());
     }
 
+    private void initRepoWithTag() throws Exception {
+        m.hg(repo, "init");
+        m.touchAndCommit(repo, "init");
+        m.hg(repo, "branch", "stable");
+        m.touchAndCommit(repo, "stable commit");
+        m.hg(repo, "tag", "release");
+        m.hg(repo, "update", "--clean", "default");
+    }
+
+    @Test public void testGetBranchFromTag() throws Exception {
```

Comment:
BTW for tracking purposes, annotate your tests with `@Issue("JENKINS-10706")` (or `@Bug(10706)` if `Issue` is not yet in the classpath, I cannot remember offhand).